www/MobileServices.Web.js

```diff
@@ -10851,7 +10865,102 @@
 		
 		    return this._push._unregisterAll(deviceToken);
 		};
-		
+		wns.prototype.registerNative = function (deviceToken, tags) {
+		    /// <summary>
+		    /// Register for native notification
+		    /// </summary>
+		    /// <param name="deviceToken">
+		    /// The deviceToken to register
+		    /// </param>
+		    /// <param name="tags" mayBeNull="true">
+		    /// Array containing the tags for this registeration
+		    /// </param>
+		    /// <returns>
+		    /// Promise that will complete when the register is completed
+		    /// </returns>
+
+		    return this._push._register('wns', deviceToken, tags);
+		};
+
+		wns.prototype.registerTemplate = function (deviceToken, name, bodyTemplate, expiryTemplate, tags) {
+		    /// <summary>
+		    /// Register for template notification
+		    /// </summary>
+		    /// <param name="deviceToken">The deviceToken to register</param>
+		    /// <param name="name">The name of the template</param>
+		    /// <param name="bodyTemplate">
+		    /// String or json object defining the body of the template register
+		    /// </param>
+		    /// <param name="expiryTemplate">
+		    /// String defining the datatime or template expresession that evaluates to a date time
+		    /// string to use for the expiry of the message
+		    /// </param>
+		    /// <param name="tags">
+		    /// Array containing the tags for this registeration
+		    /// </param>
+		    /// <returns>
+		    /// Promise that will complete when the unregister is completed
+		    /// </returns>
+
+		    if (_.isNull(tags) && !_.isNull(expiryTemplate) && Array.isArray(expiryTemplate)) {
+		        tags = expiryTemplate;
+		        expiryTemplate = null;
+		    }
+
+		    Validate.notNullOrEmpty(name, 'name');
+		    Validate.notNullOrEmpty(bodyTemplate, 'bodyTemplate');
+
+		    var templateAsString = bodyTemplate,
+		        registration = makeCoreRegistration(deviceToken, "wns", tags);
+
+		    if (!_.isString(templateAsString)) {
+		        templateAsString = JSON.stringify(templateAsString);
+		    }
+
+		    registration.headers = { "X-WNS-Type": "wns/toast" };
```

Comment:
//GMc bug fix to register as template.
            registration.templateName = name;
            registration.templateBody = templateAsString;